## Importing necessary libraries

In [1]:
# Libraries to help with reading and manipulating data
import pandas as pd
import numpy as np

# Libraries to help with data visualization
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

# Library to help with statistical analysis
import scipy.stats as stats

## Sampling Distribution

**Q1. Suppose an automobile battery manufacturer claims that the mean lifetime of their battery is 60 months with a standard deviation of 6 months. Suppose the distribution of battery life is approximately normal. Find the probability that the mean lifetime of 40 randomly sampled batteries will be less than 58 months.**

In [2]:
# import the required function
from scipy.stats import norm
# declare the value of mean lifetime of battery in mu
mu = 60
# declare the value of standard deviation of battery
sigma = 6
# sample size
n = 40
# find the sample standard deviation
s = sigma/np.sqrt(40)
# find the probability
round(norm.cdf(58, loc = mu, scale = s), 4)

0.0175

**Insight:**

* There is less than 2% chance that the mean lifetime of 40 randomly sampled batteries will be less than 58 months.

## Interval Estimation

**Q2. A random sample of 40 households was selected as part of a study on electricity usage, and the number of kilowatt-hours (kWh) was recorded for each household in the sample for the first quarter of 2020. The average usage was found to be 310 kWh. In a very large study in the first quarter of the previous year, it was found that the standard deviation of the usage was 89 kWh.**

**Assuming the standard deviation is unchanged and that the usage is normally distributed, provide an expression for calculating a 95% confidence interval for the mean usage in the first quarter of 2019.**

In [3]:
#import the required function
from scipy.stats import norm

#set the values of sample mean and sigma
x_bar, sigma = 310, 89

# set the value of sample size
n = 40

# construct the confidence interval
np.round(norm.interval(0.95, loc = x_bar, scale = sigma/np.sqrt(n)), 2)

array([282.42, 337.58])

**Insight:**

* We are 95% confident that the mean usage in the first quarter of 2019 lies between 282.42 and 337.58 kWh.

## Hypothesis Testing

**Q3. You are a manager of a Chinese restaurant. You want to determine whether the mean waiting time to place an order has changed in the past month from its previous population mean value of 4.5 minutes. State the null and alternative hypotheses.**

The null hypothesis is that the mean waiting time has not changed from its previous value of 4.5 minutes. This is stated
as

$$H_0: \mu = 4.5$$

The alternative hypothesis is that the mean waiting time has been changed from its previous value of 4.5 minutes. This is stated as

$$H_a: \mu \neq 4.5$$

**Q4. What is the p-value in a two-tailed z-test for one sample, where the computed test statistic (z-stat) is equal to +2.00?**

For calculating the p-value for a two-tailed hypothesis test, first we will calculate the p-value for a one-tailed test and then multiply the p-value by 2 to obtain the result for a two-tailed test

The p-value for a one-tailed test is the area to the right of the computed test statistic in the distribution of the test statistic. That is, the probability $P(X>z\_stat)$. This can be computed using the `cdf` function.
  * The function `norm.cdf(x, mu, sigma)` calculates the probability $P(X<x)$ or $P(X<=x)$

In [4]:
#import the required function
from scipy.stats import norm
# provided value of the test statistic is +2.00
z_stat = 2
# calculating the p-value for a one-tailed test
p_val = 1 - norm.cdf(2)
p_val

0.02275013194817921

In [5]:
# calculating the p-value for a two-tailed test
print(p_val*2)

0.04550026389635842


**Insight:**
* in a two-tailed z-test for one sample, where the computed test statistic (z-stat) is equal to +2.00, the p-value is 0.0455

**Q5. Samy, Product Manager of K2 Jeans, wants to launch a product line into a new market area. A Survey of a random sample of 400 households in that market showed a mean income per household of 30000 rupees. The standard deviation based on an earlier pilot study of households is 8000 rupees. Samy strongly believes the product line will be adequately profitable only in markets where the mean household income is greater than 29000 rupees. Samy wants our help in deciding whether the product line should be introduced in the new market. Perform statistical analysis and based on that draw a conclusion.**

Assume a level of significance ($\alpha$) of 5%

Null Hypothesis: Mean income of the household is less than or equal to 29000 rupees. This can be written as:

$$H_0: \mu \leq 29000$$

Alternative Hypothesis: Mean income of the household is greater than 29000 rupees. This can be written as:

$$H_a: \mu > 29000$$

From the description in the question, we are provided:

$\bar{x} = 30000$

$\mu = 29000$

$\sigma = 8000$

$n = 400$

The formula for computing the test statistic (z-stat) is given as:

<font size = 5> $z = \frac{(\bar{x}-\mu)}{\sigma/\sqrt{n}}$ </font>

In [6]:
# calculating the test statistic (z-stat)
z = (30000 - 29000)/(8000/np.sqrt(400))
print(z)

2.5


As the alternative hypothesis contains $>$ sign, it is a one-tailed test of greater than type. So, the p-value will be the area to the right of the computed test statistic in the distribution of the test statistic. That is, the probability $P(X>z)$

In [7]:
# Calculating the p-value for the test statistic z = 2.5
p_val = 1 - stats.norm.cdf(z)
p_val

0.006209665325776159

**Conclusion based on p-value**

* As the p-value is less than $\alpha$ (=0.05), we have enough evidence to reject the null hypothesis.

* Hence, we have enough evidence to conclude that the mean income of the household is greater than 29000 rupees